### PR TITLE
fix: improve performance of `getReport()`

### PR DESCRIPTION
**Pull request by styfle**
This will improve performance for newer versions of Node.js because this report doesn't need to access the network.

- See https://github.com/nodejs/node/pull/51645

### CODE REVIEW

File: lib/process.js
Comment (lovell) [inline lib/process.js:12]:
```suggestion
    /* istanbul ignore next */
    if (isLinux() && process.report) {
```

File: lib/process.js
Comment (lovell) [inline lib/process.js]:
```suggestion
```

**Review by lovell — COMMENTED** [review]
Thank you for the PR Steven, I hadn't seen this new feature. Is it just me or is the API we have to use to set `excludeNetwork` somewhat awkward?

I've made a suggestion to move the `/* istanbul ignore next */` exclusion up a level to the surrounding conditional, which I think should deal with the unit test failure.

~~I need to fix up the integration tests first as NodeSource have totally changed the way their repos work, plus a couple of the more _rolling_ flavours of Linux such as Fedora 39 have upgraded their version of glibc within the last few months. Leave it with me.~~

UPDATE: Please can you rebase against `main` and the integration tests should pass again.

**Comment by styfle** [comment]
> Is it just me or is the API we have to use to set `excludeNetwork` somewhat awkward?

Agreed. Its not just `excludeNetwork` thats awkward, but every Report configuration: https://nodejs.org/api/report.html#configuration

**Comment by styfle** [comment]
I'm also wondering if we should update CI to run against Node.js Nightly just to confirm this doesn't cause any problems (since this feature hasn't landed on the stable channel yet)

https://nodejs.org/download/nightly/v22.0.0-nightly20240304b34512e38e/

**Review by lovell — APPROVED** [review]

**Comment by lovell** [comment]
Thanks for the updates. Let's wait until there's a proper, non-nightly (backported?) release of Node.js with the new flag before merging/releasing.

**Comment by lovell** [comment]
I'm going to merge this now as the change in #22 means the use of `getReport()` should now mostly be confined to musl-based Linux.

There are no nightly Node.js builds for musl and the upstream `excludeNetwork` patch isn't planned for backport, so when Node.js 22 is released we can add it to the test matrix.

**Comment by lovell** [comment]
Thanks again for the PR, this change was included in v2.0.3.

**Pull request merged**
